### PR TITLE
Incorrect generalization of NaN

### LINKED ISSUE #21390: Incorrect generalization of NaN in mathematical operations
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NaN

### What specific section or headline is this issue about?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NaN#description

### What information was incorrect, unhelpful, or incomplete?

>If NaN is involved in any mathematical operation, the result is also NaN.

### What did you expect to see?

_any_ mathematical operation is too general. There are edge cases where mathematical operations involving NaN do not result in NaN. Perhaps consider replacing _any_ with _most_ or _the vast majority_

### Do you have any supporting links, references, or citations?

https://en.wikipedia.org/wiki/NaN

> NaNs can silently be removed from a chain of calculations where one calculation in the chain would give a constant result for all other floating-point values. For example, the calculation [x^0] may produce the result 1, even where x is NaN, so checking only the final result would obscure the fact that a calculation before the [x^0] resulted in a NaN.

`NaN ** 0 === 1 // true`

### Do you have anything more you want to share?

No, thank you for reviewing this issue.

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/javascript/reference/global_objects/nan`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NaN
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/nan/index.md
* Last commit: https://github.com/mdn/content/commit/5277e4fe383626b8d88a26131c133b0445f20aad
* Document last modified: 2022-10-04T03:26:59.000Z

</details>

### PR DESCRIPTION
**Pull request by GhostCore07**
Resolves #21390

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/nan/index.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ There are five different types of operations that return `NaN`:
 
 `NaN` and its behaviors are not invented by JavaScript. Its semantics in floating point arithmetic (including that `NaN !== NaN`) are specified by [IEEE 754](https://en.wikipedia.org/wiki/Double_precision_floating-point_format). `NaN`'s behaviors include:
 
-- If `NaN` is involved in any mathematical operation, the result is also `NaN`.
+- If `NaN` is involved in any mathematical operation, the result is also usually `NaN`. (See [counterexample](#silently-escaping-nan) below.)
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/global_objects/nan/index.md:35]:
```suggestion
- If `NaN` is involved in a mathematical operation, the result is usually also `NaN`. (See [counter-example](#silently_escaping_nan) below.)
```

File: files/en-us/web/javascript/reference/global_objects/nan/index.md [hunk 2]
```diff
@@ -104,6 +104,14 @@ console.log(f2b(NaN)); // Uint8Array(8) [0, 0, 0, 0, 0, 0, 248, 127]
 console.log(f2b(nan2)); // Uint8Array(8) [1, 0, 0, 0, 0, 0, 248, 127]
 ```
 
+### Silently escaping NaN
+
+In some cases simply testing for NaN at the end of a calculation is not enough to rule out the possibility that a previous calculation resulted in a NaN.
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/global_objects/nan/index.md:109]:
```suggestion
In some cases, simply testing for `NaN` at the end of a calculation is not enough to rule out the possibility that a previous calculation resulted in a `NaN`.
```

**Review by Josh-Cena — CHANGES_REQUESTED** [review]
Just curious, are there other cases where you can escape NaN? Can't think of any on top of my head

File: files/en-us/web/javascript/reference/global_objects/nan/index.md [hunk 3]
```diff
@@ -104,6 +104,14 @@ console.log(f2b(NaN)); // Uint8Array(8) [0, 0, 0, 0, 0, 0, 248, 127]
 console.log(f2b(nan2)); // Uint8Array(8) [1, 0, 0, 0, 0, 0, 248, 127]
 ```
 
+### Silently escaping NaN
+
+In some cases, simply testing for `NaN` at the end of a calculation is not enough to rule out the possibility that a previous calculation resulted in a `NaN`.
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/global_objects/nan/index.md:109]:
```suggestion
`NaN` propagates through mathematical operations, so it's usually sufficient to test for `NaN` once at the end of calculation to detect for error conditions. The only case where `NaN` gets silently escaped is when using [exponentiation](/en-US/docs/Web/JavaScript/Reference/Operators/Exponentiation) with an exponent of `0`, which immediately returns `1` without testing the base's value.
```

**Review by Josh-Cena — COMMENTED** [review]
After a quick glance over the spec, I'm convinced that this is the only special case (factoring in `Math` methods). In this case it would be interesting to note on the exponentiation operator page as well. The spec also has this to say:

> The result of _base_ `**` _exponent_ when base is 1<sub>𝔽</sub> or -1<sub>𝔽</sub> and exponent is +∞<sub>𝔽</sub> or -∞<sub>𝔽</sub>, or when base is 1<sub>𝔽</sub> and exponent is `NaN`, differs from [IEEE 754-2019](https://tc39.es/ecma262/#sec-bibliography). The first edition of ECMAScript specified a result of `NaN` for this operation, whereas later revisions of IEEE 754 specified 1<sub>𝔽</sub>. The historical ECMAScript behaviour is preserved for compatibility reasons.

Which seems interesting to note.

**Comment by Josh-Cena** [comment]
@GhostCore07 Do you intend to address the review at some point? (Because you did apply the suggestions.) If not we can still merge it and I can follow up, or I can directly commit to your branch.

**Review by GhostCore07 — COMMENTED** [review]
The changes are appropriate in my opinion.

**Comment by GhostCore07** [comment]
> @GhostCore07 Do you intend to address the review at some point? (Because you did apply the suggestions.) If not we can still merge it and I can follow up, or I can directly commit to your branch.

Sorry if I missed a step, I wasn't sure if applying the suggestions was enough or I had to also apply a comment. Since it still says 1 pending reviewer I wasn't sure if any additional action was needed by me yet.

**Comment by Josh-Cena** [comment]
Ah, I'm asking in https://github.com/mdn/content/pull/21392#pullrequestreview-1135238540 if you are willing to add this information to the `**` operator page.

**Review by Josh-Cena — APPROVED** [review]
Thanks! I've committed some changes myself. Congratulations on your first contribution here—welcome aboard!

**Pull request merged**